### ISSUE TITLE
Should add more config about `glob` to data loading , such as `dot: true`

### ISSUE THREAD

**Issue by bjwktcsnzh** [vuejs/vitepress#3599]
### Describe the bug

(Not entirely a bug , but it's annoying)

I need to add property into `glob` option param ... just like `dot: true`

https://github.com/vuejs/vitepress/blob/28989df83446923a9e7c8ada345b0778119ed66f/src/node/plugins/staticDataPlugin.ts#L102

if `dot: true` not exist , the `watchFile` list will not include the file which both on project root and start with dot , such as `.gitignore`、`.gitlab-ci` ....

### Reproduction

(Not entirely a bug)

### Expected behavior

i can add glob option param

### System Info

```Text
System:
    OS: Windows 10 10.0.19045
    CPU: (8) x64 Intel(R) Core(TM) i5-8300H CPU @ 2.30GHz
    Memory: 2.70 GB / 15.88 GB
  Binaries:
    Node: 18.18.2 - C:\Program Files\nodejs\node.EXE     
    Yarn: 1.22.19 - C:\Program Files\nodejs\yarn.CMD     
    npm: 9.8.1 - C:\Program Files\nodejs\npm.CMD
    pnpm: 8.9.2 - C:\Program Files\nodejs\pnpm.CMD
  Browsers:
    Chrome: 121.0.6167.185
    Edge: Chromium (121.0.2277.128)
    Internet Explorer: 11.0.19041.3636
  npmPackages:
    vitepress: ^1.0.0-rc.31 => 1.0.0-rc.33
```

### Additional context

The `dot` param description in fast-glob :

```
    /**
     * Allow patterns to match entries that begin with a period (`.`).
     *
     * @default false
     */
    dot?: boolean;
```

### Validations

- [X] Check if you're on the [latest VitePress version](https://github.com/vuejs/vitepress/releases/latest).
- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by bjwktcsnzh** [comment]
This is an annoying parameter of fast-glob 💢

**Comment by bjwktcsnzh** [comment]
Of course, this is not a big worry. Because I can bypass this bug without using watch mode.

**Comment by bluwy** [comment]
Maybe we can set `dot: true` by default instead

**Comment by brc-dd** [comment]
This was fixed in #4808. You can return `options: { globOptions: { dot: true } }` from the default export. Use `defineLoader` for typings.

Regarding making `dot: true` default, I think people might be using `watchedFiles` array directly. It might leak some stuff if they haven't added guard to filter out stuff.

Also, I think even without `dot: true`, you should be able to match if you explicitly write the dot version of the glob in the `watch` array (for example, similar to how we've to do in tsconfig).
